kreuzberg/_entity_extraction.py

```diff
@@ -75,29 +56,36 @@ def extract_entities(
                 )
                 for ent in results
             )
-        except (ImportError, RuntimeError):
+        except RuntimeError:
             pass
     return entities
 
 
 def extract_keywords(
     text: str,
     keyword_count: int = 10,
+    model: Any = None,
 ) -> list[tuple[str, float]]:
     """Extract keywords from text using the KeyBERT model.
 
     Args:
         text: The input text to extract keywords from.
         keyword_count: Number of top keywords to return. Defaults to 10.
+        model: Pre-initialized KeyBERT model instance. If None, a default model is used if available.
 
     Returns:
         list[tuple[str, float]]: A list of tuples containing keywords and their relevance scores.
     """
-    if KeyBERT is None:
-        return []
+    kw_model = model
+    if not kw_model:
+        try:
+            from keybert import KeyBERT
+
+            kw_model = KeyBERT()
+        except (ImportError, RuntimeError):
```

Comment:
see how we use MissingDependencyError.create_for_package -> this is what should be raised if someone tries to use this feature without the dependencies.